src/wallet/wallet.cpp

```diff
@@ -1339,11 +1339,14 @@ void CWallet::MarkConflicted(const uint256& hashBlock, int conflicting_height, c
 {
     LOCK(cs_wallet);
 
-    int conflictconfirms = (m_last_block_processed_height - conflicting_height + 1) * -1;
     // If number of conflict confirms cannot be determined, this means
     // that the block is still unknown or not yet part of the main chain,
     // for example when loading the wallet during a reindex. Do nothing in that
     // case.
+    if (m_last_block_processed_height < 0 || conflicting_height < 0) {
```

Comment:
Do you know when `conflicting_height` can be negative? It looks like a different bug if that happens.

As far as I can see, we only set the `TxStateConflicted` state at `AddToWalletIfInvolvingMe`, where `confirmed_block_height` is always positive. In `LoadToWallet`, we take the conflicting height of the prev tx, so it should never be negative.

Reply:
When the `CWalletTx` is loaded, it only contains the hash of conflicting block. The height will be set to -1. `LoadToWallet` will try to lookup the block by hash, but if there is not chain available (as with the wallet tool and during migration), then it cannot do that. So the height will remain -1. If that transaction has a child, and that child is loaded after the parent is loaded, then `MarkConflicted` will be called using the parent's in-memory conflicted state. Since there was no chain, the `conflicting_height` that is passed to `MarkConflicted` is -1, hence triggering this error.